functorch/csrc/BatchRulesScatterOps.cpp

```diff
@@ -179,16 +179,20 @@ namespace {
                               ArrayRef<optional<Tensor>> indices,
                               ArrayRef<optional<int64_t>> indices_bdims,
                               const Tensor &values,
-                              optional<int64_t> values_bdim) {
+                              optional<int64_t> values_bdim,
+                              optional<int64_t> opt_batch_size = {}) {
 
-    auto self_logical_rank = rankWithoutBatchDim(self, self_bdim);
-    auto values_logical_rank = rankWithoutBatchDim(values, values_bdim);
     Tensor self_ = moveBatchDimToFront(self, self_bdim);
     Tensor values_ = moveBatchDimToFront(values, values_bdim);
-    const auto batch_size = self_.size(0);
+    // for inplace variants `index_put_` and `_index_put_impl_` we find the batch_size
+    // here while for `index_put` does it outside of this function.
+    const auto batch_size = opt_batch_size ? opt_batch_size.value() : self_.size(0);
+    self_ = ensure_has_bdim(self_, self_bdim.has_value(), batch_size);
     values_ = ensure_has_bdim(values_, values_bdim.has_value(), batch_size);
     TORCH_INTERNAL_ASSERT(indices.size() == indices_bdims.size());
-    std::vector<optional<Tensor>> indices_ = batchIndices(indices, indices_bdims, self_.size(0), self_bdim, values_bdim);
+
+    // we make sure that `self` has bdim at `0`.
```

Comment:
nit: "we've already made sure that `self` has bdim at `0`" to clarify that we've already done this (the way this is currently written is ambiguous: did we already do this, or does this happen in batchIndices?)